clients/src/main/java/org/apache/kafka/clients/consumer/RoundRobinAssignor.java

```diff
@@ -61,23 +61,25 @@
 
     @Override
     public Map<String, List<TopicPartition>> assign(Map<String, Integer> partitionsPerTopic,
-                                                    Map<String, Subscription> subscriptions) {
+                                                    Map<MemberInfo, Subscription> subscriptions) {
         Map<String, List<TopicPartition>> assignment = new HashMap<>();
-        for (String memberId : subscriptions.keySet())
-            assignment.put(memberId, new ArrayList<>());
+        for (MemberInfo memberInfo : subscriptions.keySet()) {
+            assignment.put(memberInfo.memberId, new ArrayList<>());
+        }
+
+        CircularIterator<MemberInfo> assigner = new CircularIterator<>(Utils.sorted(subscriptions.keySet()));
```

Comment:
This is a meta comment: I'd like to include in this PR

1) doc changes illustrating what behavioral changes we are making for round robin assignor (similarly for future PRs where we change for range assignor).

2) and javadoc change that state under which conditions, leveraging on instance id would help. My understanding is that it must be 2.a) same num.members compared with the previous generation and 2.b) all static members stay the same, no more static members joining and no stat members exiting in this new generation.